### PR TITLE
Create users via the API

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-363

Not ready for review yet.

### CODE REVIEW

**Comment by MartinEden** [comment]
Okay, this is a big one. I think we should probably pair on reviewing it.

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqUserRepository.kt [hunk 1]
```diff
@@ -76,6 +77,15 @@ class JooqUserRepository(db: JooqContext) : JooqRepository(db), UserRepository
...
     }
 
+    override fun addUser(user: CreateUser)
+    {
+        dsl.newRecord(APP_USER).apply {
+            username = user.username
+            name = user.name
+            email = user.email
+        }.insert()
+    }
```
Comment (littledice) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqUserRepository.kt]: double check that a user with no pw cannot log in (add unit test)

File: src/app/src/main/resources/config.properties [hunk 2]
```diff
@@ -10,3 +10,4 @@ token.issuer=${token_issuer}
 
 app.version=${app_version}
 app.port=${app_port}
+app.url=${app_url}
```
Comment (littledice) [inline src/app/src/main/resources/config.properties]: can you get this from the request?

File: src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/CreateUserTests.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            "email" to "gandalf@example.com"
+    )
+
+    @Test
+    fun `can create user`()
+    {
+        val objectUrl = "/users/gandalf.grey/"
+
+        validate("/users/", HttpMethod.post) against "CreateUser" sending {
+            postData.toJsonObject()
```
Comment (littledice) [inline src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/CreateUserTests.kt]: can we avoid overloading 'against' here

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqUserRepository.kt (hunk 1 above)
Comment (MartinEden) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqUserRepository.kt]: Fixed

File: src/app/src/main/resources/config.properties (hunk 2 above)
Comment (MartinEden) [inline src/app/src/main/resources/config.properties]: It turns out yes, but as I suspected it gets messed up by the reverse proxy.

File: src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/CreateUserTests.kt (hunk 3 above)
Comment (MartinEden) [inline src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/CreateUserTests.kt]: Replaced it with `withRequestSchema`.

**Comment by MartinEden** [comment]
Ready for review again.

**Pull request merged**